python/ray/util/client/server/proxier.py

```diff
@@ -191,13 +197,18 @@ def start_specific_server(self, client_id: str,
 
         serialized_runtime_env = job_config.get_serialized_runtime_env()
 
+        output, error = self.node.get_log_file_handles(
+            f"ray_client_server_{specific_server.port}", unique=True)
+
         proc = start_ray_client_server(
-            self._get_redis_address(),
+            self.redis_address,
             specific_server.port,
+            stdout_file=output,
+            stderr_file=error,
             fate_share=self.fate_share,
             server_type="specific-server",
             serialized_runtime_env=serialized_runtime_env,
-            session_dir=self._get_session_dir())
+            session_dir=self.node._session_dir)
```

Comment:
Do you want to use the API here instead of accessing the internal variables?

https://github.com/ray-project/ray/blob/beefcf30500c6e3a0e79baa52f2d9bc21c3a37b2/python/ray/node.py#L466-L468

Reply:
Great callout! I will switch to this!